hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -441,6 +441,8 @@ def __init__(self, elements):
         assert self.elements
 
     def __repr__(self):
+        if isinstance(self.elements, range):
+            return "sampled_from(%r)" % (self.elements,)
```

Comment:
I kinda feel this should just be our default? It's not totally clear to me why we have the repr that pretends `sampled_from` was called with a list at all.

Reply:
IMO `sampled_from([1, 2, 3])` is nicer than the alternative `sampled_from((1, 2, 3))` - it makes sense to have `check_sample` return a tuple to prevent accidental mutation, but I also prefer to have contrasting brackets in the repr.

Reply:
We could reverse the default?

```
if isinstance(self.elements, tuple):
    return "sampled_from([%s])" % (repr(self.elements)[1:-1],)
else:
   return "sampled_from(%r)" % (self.elements,)
```

That way rather than special casing everything we want to have a normal repr, we special case the one thing we'd prefer not to show.